packages/sdk/src/index.ts

```diff
@@ -379,16 +383,29 @@ export default class SDK {
     ) {
       log.verbose("node is not reliable enough. selecting new entry node");
       exclusionList.push(this.entryNode!.peerId);
+      // Try to select entry node 3 times
       try {
-        await this.selectEntryNode(
-          this.ops.discoveryPlatformApiEndpoint,
-          exclusionList
+        await retry(
```

Comment:
since `createRequest` can be called multiple times at the same instance, this `retry` will be triggered multiple times, maybe we need to check if `this.selectingEntryNode` is `true`?

Reply:
At the start of the `createRequest` function there is a check for `this.selectingEntryNode` which avoids the retry being called multiple times.